snapcraft/plugins/rust.py

```diff
@@ -130,6 +165,12 @@ def clean_pull(self):
         if os.path.exists(self._rustpath):
             shutil.rmtree(self._rustpath)
 
+    def clean_build(self):
+        super().clean_build()
+
+        with suppress(FileExistsError):
+            shutil.rmtree(self._cargo_dir)
```

Comment:
Wait... why would `rmtree` raise `FileExistsError`? If a file is in use? That sounds like a situation where we shouldn't suppress. Curious what's happening here.